roxctl/deployment/check/check.go

```diff
@@ -116,6 +117,13 @@ func Command(cliEnvironment environment.Environment) *cobra.Command {
 	c.Flags().BoolVar(&deploymentCheckCmd.force, "force", false, "bypass Central's cache for images and force a new pull from the Scanner")
 	utils.Must(c.MarkFlagRequired("file"))
 	c.Flags().StringVar(&deploymentCheckCmd.cluster, "cluster", "", "cluster name or ID to use as context for evaluation")
+	c.Flags().StringVarP(&deploymentCheckCmd.namespace, "namespace", "n", "default", "namespace to evaluate, defaults to \"default\" if unset.")
+	namespace, err := c.Flags().GetString("namespace")
```

Comment:
You usually do not have access to the value of this flag at this point in code, as it has not been set by the user yet.
You can call this line the earliest inside of `RunE: func(cmd *cobra.Command, args []string) error {`

Reply:
Maybe this is part of the positioning of the check, but it is also currently triggering on every command, which might not be the intention:
```
$ bin/linux_amd64/roxctl version

Warning: The namespace defaulted to "default", if you want to check deployments in a custom namespace set it with 'default [X]' or '-n [X]'
4.3.x-451-g2714d8c1eb

```

Reply:
I should've tested this. I've moved the default check and warning to the Construct() function where flag values have been set.